sdk/js/src/relayer/relayer/helpers.ts

```diff
@@ -385,23 +506,75 @@ export async function getDeliveryHash(
         index > 0 ? ` (the ${index}-th wormhole relayer log was requested)` : ""
       }`
     );
-  const log = logs[index];
+  return getDeliveryHashFromLog(
+    logs[index],
+    CHAINS[sourceChain],
+    provider,
+    rx.blockHash
+  );
+}
+
+export async function getDeliveryHashFromLog(
+  wormholeLog: ethers.providers.Log,
+  sourceChain: ChainId,
+  provider: ethers.providers.Provider,
+  blockHash: string
+): Promise<string> {
   const wormholePublishedMessage =
-    Implementation__factory.createInterface().parseLog(log);
-  const block = await provider.getBlock(rx.blockHash);
-  const body = ethers.utils.solidityPack(
-    ["uint32", "uint32", "uint16", "bytes32", "uint64", "uint8", "bytes"],
+    Implementation__factory.createInterface().parseLog(wormholeLog);
 
-    [
-      block.timestamp,
-      wormholePublishedMessage.args["nonce"],
-      CHAINS[sourceChain],
-      log.topics[1],
-      wormholePublishedMessage.args["sequence"],
-      wormholePublishedMessage.args["consistencyLevel"],
-      wormholePublishedMessage.args["payload"],
-    ]
+  const block = await provider.getBlock(blockHash);
+
+  return getDeliveryHashFromVaaFields(
+    sourceChain,
+    wormholeLog.topics[1],
+    wormholePublishedMessage.args["sequence"],
+    block.timestamp,
+    wormholePublishedMessage.args["nonce"],
+    wormholePublishedMessage.args["consistencyLevel"],
+    wormholePublishedMessage.args["payload"]
   );
-  const deliveryHash = ethers.utils.keccak256(ethers.utils.keccak256(body));
-  return deliveryHash;
+}
+
+export async function getCCTPMessageLogURL(
+  cctpKey: CCTPKey,
+  sourceChain: ChainName,
+  receipt: ethers.providers.TransactionReceipt,
+  environment: Network
+) {
+  let cctpLog;
+  let messageSentLog;
+  const DepositForBurnTopic =
+    "0x2fa9ca894982930190727e75500a97d8dc500233a5065e0f3126c48fbe0343c0";
+  const MessageSentTopic =
+    "0x8c5261668696ce22758910d05bab8f186d6eb247ceac2af2e82c7dc17669b036";
```

Comment:
can you get these from the contract itself? https://github.com/wormhole-foundation/connect-sdk/blob/develop/platforms/evm/src/protocols/circleBridge.ts#L61-L69

Reply:
Added clarification comments; the constants are just // keccak256("DepositForBurn(uint64,address,uint256,address,bytes32,uint32,bytes32,bytes32)")
  // keccak256("MessageSent(bytes)")

I could get them from the contract but would require me to bring the Circle contract ABIs into this SDK too - so perhaps this is an ok intermediate solution (explanatory comments explaining how the constants are calculated!)

Reply:
nvm - replaced the constants with explicitly keccak-ing them!